blueocean-dashboard/src/main/js/components/records.jsx

```diff
@@ -23,7 +23,7 @@ export class PipelineRecord extends Record({
     weatherScore: 0,
 }) {
     isFolder() {
-        return this.numberOfFolders !== null;
+        return this._class === 'io.jenkins.blueocean.service.embedded.rest.PipelineFolderImpl';
```

Comment:
@michaelneale this works, but maybe we should simply keep code something like, `this.numberOfFolders !== null && totalNumberOfBranches == null` with `TODO to say replace by class/capability`, because this discovery of whats what needs to be dealt with in a common way, for example sometimes it needs to go and call /classes API to discover what other classes it knows or what other capabilities it might have. I think Keith was playing with this thing for JUnit report stuff.